go/interpreter/lfvm/memory.go

```diff
@@ -224,22 +212,30 @@ func (m *Memory) CopyData(offset uint64, trg []byte) {
 	}
 }
 
-func (m *Memory) GetSlice(offset, size uint64) []byte {
-	if size == 0 {
-		return nil
+// readSliceAndExpandMemory reads size bytes from the memory at the given offset.
+// Expands memory as needed and charges for it.
+// Returns error in case of not enough gas or offset+32 overflow.
+func (m *Memory) readSliceAndExpandMemory(offset, size uint64, c *context) ([]byte, error) {
```

Comment:
I would associate a `read` prefix with a function to which I can pass the target of the read operation, like it is done for the `readWord` function below, or in Go's [Reader](https://pkg.go.dev/io#Reader) interface.

Something that returns me a part of some state is a valid `getSlice`. 

The `AndExpandMemory` could also be dropped since there is no `readSlice` or `getSlice` counterpart.